src/vs/editor/contrib/wordHighlighter/browser/textualHighlightProvider.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { USUAL_WORD_SEPARATORS } from 'vs/editor/common/core/wordHelper';
+import { ILanguageFeaturesService } from 'vs/editor/common/services/languageFeatures';
+import { DocumentHighlight, DocumentHighlightKind, MultiDocumentHighlightProvider, ProviderResult } from 'vs/editor/common/languages';
+import { ITextModel } from 'vs/editor/common/model';
+import { Position } from 'vs/editor/common/core/position';
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { ResourceMap } from 'vs/base/common/map';
+
+
+class TextualDocumentHighlightProvider implements MultiDocumentHighlightProvider {
+
+	provideMultiDocumentHighlights(primaryModel: ITextModel, position: Position, otherModels: ITextModel[], token: CancellationToken): ProviderResult<ResourceMap<DocumentHighlight[]>> {
+
+		const result = new ResourceMap<DocumentHighlight[]>();
+
+		const word = primaryModel.getWordAtPosition({
+			lineNumber: position.lineNumber,
+			column: position.column
+		});
+		if (!word) {
+			return Promise.resolve(result);
+		}
+
+
+		for (const otherModel of otherModels) {
```

Comment:
Shouldn't this also include the `primaryModel`? Otherwise we highlight every except for the place in which the cursor is active, right?

Reply:
Yep. Got dropped in the refactor.

Reply:
Yep. Got dropped in the refactor.